File: new_kb/parser_samples/sample.ml
type point = { x : int; y : int }

(* adds two points *)
let add (p1 : point) (p2 : point) : point =
  { x = p1.x + p2.x; y = p1.y + p2.y }

module Sample = struct
  let origin = { x = 0; y = 0 }
end

let () =
  let result = add Sample.origin { x = 3; y = 4 } in
  Printf.printf "%d, %d\n" result.x result.y

File: new_kb/parser_samples/sample.rs
use std::fmt;

pub struct Point {
    pub x: i32,
    pub y: i32,
}

impl Point {
    pub fn new(x: i32, y: i32) -> Self {
        Point { x, y }
    }

    // adds two points
    pub fn add(&self, other: &Point) -> Point {
        Point::new(self.x + other.x, self.y + other.y)
    }
}

impl fmt::Display for Point {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        write!(f, "({}, {})", self.x, self.y)
    }
}

fn main() {
    let p1 = Point::new(1, 2);
    let p2 = Point::new(3, 4);
    println!("{}", p1.add(&p2));
}
